### PR TITLE
Use nginx-unprivileged base image

### LINKED ISSUE #48: Image runs as root
### Current Behavior:
The currently used base image runs as root:
https://github.com/DependencyTrack/frontend/blob/53a1734b7438d336367f3a80a622e9d79cd5ab73/docker/Dockerfile#L22 

### Steps to Reproduce:
Check the `Dockerfile` reference above.

### Expected Behavior:
Use the official [docker-nginx-unprivileged](https://github.com/nginxinc/docker-nginx-unprivileged) as base image.

### Environment:
  
  - Dependency-Track Version: `1.1.0`
  - Client Browser: `N/A`
  - Client O/S: `N/A`

### Additional Details:
`N/A`

### PR DESCRIPTION
**Pull request by alex1989hu**
The new images exposes 8080 port, runs as nginx
(non-root user).

There is no need to copy-paste either CMD or
ENTRYPOINT.

nginx-unprivileged configuration includes conf.d
directory, thus no need to copy-paste the
boilerplate configuration.

The custom entrypoint can be moved into existing
/docker-entrypoint.d/ directory.

Use the same `--chown` strategy as already used in
Dockerfile made for DependencyTrack/dependency-track.

Fixes #48 

Signed-off-by: Alex Szakaly <alex.szakaly@gmail.com>

### CODE REVIEW

**Comment by alex1989hu** [comment]
Update is needed for `docker-compose`, too.

**Comment by Otterian** [comment]
Would be nice to have this merged soon :+1:

**Comment by stevespringett** [comment]
Thanks for the PR. I'll test it out this week.

**Comment by davidkarlsen** [comment]
@stevespringett bump. This also blocks the path forward for the helm-chart, which is the preferred way to run DT in k8s.

File: docker/Dockerfile [hunk 1]
```diff
@@ -19,24 +19,25 @@ RUN npm run build
...
+USER 101
 
 # Set default settings that may get overridden to empty values by
 # the entrypoint script, if not explicitly provided by the user
 ENV OIDC_SCOPE "openid profile email"
 
 # Setup entrypoint
 WORKDIR /app
-COPY ./docker/docker-entrypoint.sh /usr/bin/docker-entrypoint.sh
-ENTRYPOINT [ "/usr/bin/docker-entrypoint.sh" ]
```
Comment (stevespringett) [inline docker/Dockerfile:38]: Unsure why entrypoint was removed. Can you clarify?

File: docker/Dockerfile [hunk 2]
```diff
@@ -19,24 +19,25 @@ RUN npm run build
...
-
-# Inherited from parent image
-# See https://github.com/nginxinc/docker-nginx/blob/1.18.0/stable/alpine/Dockerfile#L115
-CMD ["nginx", "-g", "daemon off;"]
```
Comment (stevespringett) [inline docker/Dockerfile:42]: Unsure why cmd was removed. Can you clarify?

**Review by stevespringett — COMMENTED** [review]
Thanks for the PR. Can you clarify the few inline questions please?

File: docker/Dockerfile (hunk 1 above)
Comment (davidkarlsen) [reply docker/Dockerfile:38]: https://github.com/DependencyTrack/frontend/pull/49#issue-548201761

File: docker/Dockerfile (hunk 2 above)
Comment (davidkarlsen) [reply docker/Dockerfile:42]: https://github.com/DependencyTrack/frontend/pull/49#issue-548201761

File: docker/Dockerfile (hunk 1 above)
Comment (stevespringett) [reply docker/Dockerfile:38]: Thanks David. I missed that part.

File: docker/Dockerfile (hunk 2 above)
Comment (stevespringett) [reply docker/Dockerfile:42]: Thanks David. I missed that part.

**Comment by stevespringett** [comment]
I may have to revert. These changes broke the image and I'm trying to get v4.1 released

```
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d//docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-oidc-configuration.sh/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh/docker-entrypoint.sh: Configuration complete; ready for start up
```

**Comment by alex1989hu** [comment]
> I may have to revert. These changes broke the image and I'm trying to get v4.1 released
> 
> ```
> /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d//docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-oidc-configuration.sh/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh/docker-entrypoint.sh: Configuration complete; ready for start up
> ```

The image is not broken, no rush. The image can serve traffic. The built-in ipv6 script is not the best solution, you can ugnore it or remove from the image.

**Pull request merged**